From d0550de19cacab3e591641ba358a72fbc798b231 Mon Sep 17 00:00:00 2001
From: Dave Airlie <airlied@redhat.com>
Date: Mon, 22 Sep 2014 12:45:02 +1000
Subject: [PATCH] sis: fix build against latest xserver

---
 src/sis.h     | 2 ++
 src/vstruct.h | 2 ++
 2 files changed, 4 insertions(+)

diff --git a/src/sis.h b/src/sis.h
index 20e6134..385b784 100644
--- a/src/sis.h
+++ b/src/sis.h
@@ -1004,7 +1004,9 @@ typedef struct {
     ScrnInfoPtr		pScrn;
     pciVideoPtr		PciInfo;
     int			PciBus, PciDevice, PciFunc;
+#ifndef XSERVER_LIBPCIACCESS
     PCITAG		PciTag;
+#endif
     EntityInfoPtr	pEnt;
     int			Chipset;
     unsigned char	ChipType;
diff --git a/src/vstruct.h b/src/vstruct.h
index 88f7eb6..5d7a98d 100644
--- a/src/vstruct.h
+++ b/src/vstruct.h
@@ -237,8 +237,10 @@ struct SiS_Private
 	unsigned char			ChipType;
 	unsigned char			ChipRevision;
 #ifdef SIS_XORG_XF86
+#ifndef XSERVER_LIBPCIACCESS
 	PCITAG				PciTag;
 #endif
+#endif
 #ifdef SIS_LINUX_KERNEL
 	void				*ivideo;
 #endif
-- 
1.8.4.5

